### ISSUE TITLE
/home/core/workspace/aheuermann/test/deploykey_rsa: No such file or directory

### ISSUE THREAD

**Issue by aheuermann** [groupon/DotCi#179]
First of all, thanks for the great project!
I recently ran into the following error when the sub-builds for a private repo run on a different slave node than the parent:
```bash
Building remotely on EC2 Pull Request Validator Instance (PullRequestValidator) in workspace /home/core/workspace/aheuermann/test/script=node
[script=node] $ /bin/bash -le /tmp/hudson7614165341005502056.sh
$  chmod -R u+w . ; find . ! -path "./deploykey_rsa.pub" ! -path "./deploykey_rsa" -delete
$  git init
Initialized empty Git repository in /home/core/workspace/aheuermann/test/script=node/.git/
$  git remote add origin git@github.com:aheuermann/test.git
$  ssh-agent bash -c "ssh-add -D && ssh-add \/home/core/workspace/aheuermann/test/deploykey_rsa && git fetch origin dotci "
All identities removed.
/home/core/workspace/aheuermann/test/deploykey_rsa: No such file or directory
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```
I think I can workaround this by increasing the number of executors on each slave node and prevent other builds from running in parallel on the same node.

**Comment by suryagaddipati** [comment]
@aheuermann thank you for reporting this. 

According to [code](https://github.com/groupon/DotCi/blob/master/src/main/java/com/groupon/jenkins/dynamic/build/DynamicSubBuild.java#L134) looks like deploy keys are being exported into the repo before running the build.  A little puzzled why they might be missing .


I can investigate this more.

**Comment by aheuermann** [comment]
Note that it is looking in the parent's workspace.  Should it be looking in the workspace of the current sub-build?

**Comment by suryagaddipati** [comment]
Ah..def looks like a bug.

**Comment by aheuermann** [comment]
@suryagaddipati have you had a chance to look at this? If not, I will see if I can track it down.

**Comment by suryagaddipati** [comment]
Fixed in (2.19.3)[http://maven.jenkins-ci.org:8081/content/repositories/releases/com/groupon/jenkins-ci/plugins/DotCi/2.19.3/].
